krux_marathon_api/marathonapi.py

```diff
@@ -107,86 +96,70 @@ def assign_config_data(self, config_file_data, marathon_app_result):
         marathon_app_result.user                     = config_file_data["user"]                 #updates confirmed
 
     def list_marathon_apps(self, marathon_server):
+        """
+        This method connects to the marathon server and requests all the
+        Marathon Apps.
+        """
         self.logger.info("Listing apps running on marathon")
         current_marathon_apps = marathon_server.list_apps()
-        self.logger.info(current_marathon_apps)
-        print(current_marathon_apps)
-
-    def get_marathon_app(self, marathon_server, config_file_data):
-        self.logger.info("Getting app info from the marthon server")
-        if self.marathon_config_file:
+        return current_marathon_apps
+
+    def get_marathon_app(self, marathon_server, config_file_data, app_id):
+        """
+        Connects to the Marathon Server using the supplied App ID.
+        If we passed the config file and the App isn't found on the server, we
+        try to create it.
+        If the App name was passed on the command line instead, we will just
+        return with an error if the App wasn't found.
```

Comment:
Nit: This logic _feels_ weird. `config_file_data` determines whether an error is thrown or not? I would rather have this return `None` when there isn't any app under that name and have a create method. Alternatively, you can make this method return an marathon app object that can later be saved using a save method. Feel free to ask more details about these two approaches.